firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,262 @@
+/* chunks
+
+Scenario: A ChunkSplitter constructed with default parameters correctly handles input chunks of acceptable lengths
+
+  GIVEN("A char chunk splitter with an empty internal buffer of 102 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("102 non-delimiter bytes are passed as input, and 103rd byte is passed as a delimiter, and output is called between each input")
+      THEN("The input method reports ok status for all the 102 non-delimeter bytes")
+      THEN("After the input of 102 non-delimiter bytes, The output method reports waiting status")
+      THEN("The input method reports output_ready status on the 103rd byte")
+      THEN("After the delimiter is passed to input, The output method reports ok status")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A uint8_t chunksplitter with an empty internal buffer of capacity 256 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("A delimiter is passed as input and output method is called after that")
+      THEN("The input method reports output_ready status")
+      THEN("The output method reports ok status")      
+      THEN("The output buffer remains unchanged")
+
+    WHEN("255 non-delimiter bytes and 256th byte as a delimeter are passed as input, after which output is called") {      
+      THEN("The input method reports ok status for all the 255 non-delimeter bytes")
+      THEN("The input method reports output_ready status on the 256th delimeter byte")
+      THEN("After the delimiter is passed as input, The output method reports ok status")
+      THEN("The output buffer is as expected")      
+
+    WHEN("256 non-delimiter bytes are passed as input, and 257th byte is passed as a delimiter, and output is called between each input")
+      THEN("The input method reports ok status for all the 256 non-delimeter bytes")
+      THEN("The output method reports waiting status")
+      THEN("The output buffer remains unchanged")
+      THEN("The input method reports output_ready status on the 257th delimeter byte")
+      THEN("After the delimiter is passed as input, The output method reports ok status")
+      THEN("The output buffer is as expected")
+
+Scenario: A ChunkSplitter constructed with default value for delimeter and include_delimeter boolean as true correctly handles input chunks of acceptable lengths
+
+  GIVEN("A uint8_t chunksplitter with an empty internal buffer of capacity 256 bytes")
+  // assertions:
+  // output buffer is empty
+
+    WHEN("A delimiter is passed as input and output method is called after that")
+      THEN("The input method reports output_ready status")
+      THEN("The output method reports ok status")      
+      THEN("The output buffer remains unchanged")
```

Comment:
Is this correct? With `include_delimiter == true` I would expect that the output buffer should be a 1-byte buffer, consisting of the delimiter

Reply:
Ya it's wrong. its a 1-byte buffer with delimiter